### PR TITLE
DEV: Support custom icon when adding custom link to sidebar via plugin API

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/plugin-api.js [hunk 1]
```diff
@@ -1850,6 +1850,7 @@ class PluginApi {
    * @param {string} arg.name - The name of the link. Needs to be dasherized and lowercase.
    * @param {string=} arg.route - The Ember route name to generate the href attribute for the link.
    * @param {string=} arg.href - The href attribute for the link.
+   * @param {string=} arg.icon - The FontAwesome icon to display for the link.
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/lib/plugin-api.js:1853]: What does the = means here?

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/lib/plugin-api.js:1853]: Oh ok I see it's optional, I think I have seen it done differently too. This syntax is google closure syntax, JSDoc syntax is `{string} [defaultvalue]` and I guess google closure should be `{string=} defaultvalue`.

https://jsdoc.app/tags-param.html

Anyways no big deal, was just curious.

File: app/assets/javascripts/discourse/app/lib/sidebar/custom-community-section-links.js [hunk 2]
```diff
@@ -65,6 +66,14 @@ export function addSectionLink(args, secondary) {
...
         return args.title;
       }
+
+      get prefixValue() {
+        if (args.icon) {
+          return args.icon;
+        } else {
+          return super.prefixValue;
+        }
+      }
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/lib/sidebar/custom-community-section-links.js:76]: I would probably write it like this, but up to you:

```suggestion
        return args.icon || super.prefixValue;
      }
```

**Review by jjaffeux — APPROVED** [review]
:+1:

File: app/assets/javascripts/discourse/app/lib/sidebar/custom-community-section-links.js (hunk 2 above)
Comment (tgxworld) [reply app/assets/javascripts/discourse/app/lib/sidebar/custom-community-section-links.js]: :+1:

File: app/assets/javascripts/discourse/app/lib/plugin-api.js (hunk 1 above)
Comment (tgxworld) [reply app/assets/javascripts/discourse/app/lib/plugin-api.js]: O good catch. We actually want to follow JSDoc syntax here. Have made the changes :+1:

**Comment by tgxworld** [comment]
Thank you for the review @jjaffeux

**Pull request merged**